### PR TITLE
Don't add the sudo prefix if the sudo program is not on the path

### PR DESCRIPTION
**Pull request by sameagen-MW**
Also includes dependabot vulnerability updates.

### CODE REVIEW

**Comment by davidbuzinski** [comment]
Looks like this does fix one of the dependabot vulnerabilities, but the rest are in the files
*  tasks/install-matlab/v1/package-lock.json 
*  tasks/install-matlab/v0/package-lock.json
*  tasks/run-matlab-command/v0/package-lock.json
*  tasks/run-matlab-build/v0/package-lock.json
*  tasks/run-matlab-tests/v0/package-lock.json

Probably need to go to each of these subdirs and doing an `npm audit fix` to resolve them because of how this project is structured. It looks like the vulnerability in each of these package-locks is the same and has a breaking change so `npm audit fix` won't resolve it without `--force`.

We can handle that in a separate PR or in this one, up to you. 

Either way, the sudo changes look good!

File: tasks/install-matlab/v1/src/script.ts [hunk 1]
```diff
@@ -7,8 +7,9 @@ import * as path from "path";
 export async function downloadAndRunScript(platform: string, url: string, args: string | string[]) {
     const scriptPath = await toolLib.downloadTool(url);
     const bashPath = taskLib.which("bash", true);
+    const sudoPath = taskLib.which("sudo", false);
     let bash;
-    if (platform === "win32") {
+    if (!sudoPath || platform === "win32") {
```
Comment (mcafaro) [inline tasks/install-matlab/v1/src/script.ts:12]: Is the `platform === "win32"` necessary any more with the sudo check?

Comment (sameagen-MW) [reply tasks/install-matlab/v1/src/script.ts:12]: I think that should be droppable now, the only case in which there could be issues would be if the user had a custom `sudo.exe` somewhere on the path with different behavior than the Unix `sudo`. That seems pretty unlikely to me, so the check should be droppable.

**Comment by sameagen-MW** [comment]
WIP, still some commits left to come.

File: tasks/install-matlab/v1/src/matlab.ts [hunk 2]
```diff
@@ -70,18 +70,42 @@ async function resolveLatest(): Promise<string> {
...
 }
 
-export async function setupBatch(platform: string): Promise<void> {
-    const batchInstallDir = script.defaultInstallRoot(platform, "matlab-batch");
-    const exitCode = await script.downloadAndRunScript(platform, "https://ssd.mathworks.com/supportfiles/ci/matlab-batch/v0/install.sh", batchInstallDir);
-
-    if (exitCode !== 0) {
-        return Promise.reject(Error(`Failed to install matlab-batch. Script exited with non-zero code ${exitCode}.`));
+export async function setupBatch(platform: string, architecture: string): Promise<void> {
+    const mpmRootUrl: string = "https://ssd.mathworks.com/supportfiles/ci/matlab-batch/v1/";
```
Comment (davidbuzinski) [inline tasks/install-matlab/v1/src/matlab.ts:74]: Need to revert the changes in this file and matlab.test.ts for now

**Review by davidbuzinski — APPROVED** [review]

File: tasks/install-matlab/v0/test/suite.ts [hunk 3]
```diff
@@ -1,79 +1,67 @@
-// Copyright 2020-2022 The MathWorks, Inc.
```
Comment (mcafaro) [inline tasks/install-matlab/v0/test/suite.ts:1]: 2024

**Review by mcafaro — APPROVED** [review]

File: tasks/install-matlab/v0/test/suite.ts (hunk 3 above)
Comment (sameagen-MW) [reply tasks/install-matlab/v0/test/suite.ts:1]: Oh no, I've been still thinking it's 2023 😵
